src/overlay/Peer.cpp

```diff
@@ -294,6 +294,65 @@ Peer::getFlowControlJsonInfo() const
     return res;
 }
 
+Json::Value
+Peer::getJsonInfo(bool compact) const
+{
+    Json::Value res;
+    res["address"] = mAddress.toString();
+    res["elapsed"] = (int)getLifeTime().count();
+    res["latency"] = (int)getPing().count();
+    res["ver"] = getRemoteVersion();
+    res["olver"] = (int)getRemoteOverlayVersion();
+    res["flow_control"] = getFlowControlJsonInfo();
+    if (!compact)
+    {
+        res["message_read"] =
+            static_cast<Json::UInt64>(mPeerMetrics.mMessageRead);
+        res["message_write"] =
+            static_cast<Json::UInt64>(mPeerMetrics.mMessageWrite);
+        res["byte_read"] = static_cast<Json::UInt64>(mPeerMetrics.mByteRead);
+        res["byte_write"] = static_cast<Json::UInt64>(mPeerMetrics.mByteWrite);
+
+        res["async_read"] = static_cast<Json::UInt64>(mPeerMetrics.mAsyncRead);
+        res["async_write"] =
+            static_cast<Json::UInt64>(mPeerMetrics.mAsyncWrite);
+
+        res["message_drop"] =
+            static_cast<Json::UInt64>(mPeerMetrics.mMessageDrop);
+
+        res["max_message_delay_in_write_queue_msec"] =
+            static_cast<Json::UInt64>(
+                mPeerMetrics.mMaxMessageDelayInWriteQueue.count());
+        res["max_message_delay_in_async_write_msec"] =
+            static_cast<Json::UInt64>(
+                mPeerMetrics.mMaxMessageDelayInAsyncWrite.count());
+        res["max_outbound_queue_delay_scp_msec"] = static_cast<Json::UInt64>(
+            mPeerMetrics.mMaxOutboundQueueDelaySCP.count());
+        res["max_outbound_queue_delay_txs_msec"] = static_cast<Json::UInt64>(
+            mPeerMetrics.mMaxOutboundQueueDelayTxs.count());
+
+        res["unique_flood_bytes_recv"] =
```

Comment:
do we need all of these byte metrics? We already have a very coarse byte measure (total bytes read/written), so I wonder if only including unique and duplicate message counts is sufficient here.

Reply:
Removed from reporting. (I kept the member variables since they are used for the survey.)